core/commonMain/src/implementations/immutableList/AbstractPersistentList.kt

```diff
@@ -31,11 +22,8 @@ abstract class AbstractPersistentList<E> : PersistentList<E>, AbstractList<E>()
     }
 
     override fun removeAll(elements: Collection<E>): PersistentList<E> {
-        return mutate { it.removeAll(elements) }
-    }
-
-    override fun removeAll(predicate: (E) -> Boolean): PersistentList<E> {
-        return mutate { it.removeAll(predicate) }
+        val hashSet = elements.toHashSet()
+        return removeAll { hashSet.contains(it) }
```

Comment:
This may change behavior if `elements` collection has non-default notion of equality.

Reply:
+
Replaced `toHashSet` with `convertToSetForSetOperationWith`